### ISSUE TITLE
Assignment nodes and overlays

### ISSUE THREAD

**Issue by jrabramson** [ulfurinn/wongi-engine#49]
Hi there,

Overlays are fantastic but it seems like assignment nodes should take them into account. If you try adding an assign to that new overlay spec you'll see the test fail. I got around it by passing in the token.wme to the new token in assignment_node's beta_activate.

```
def beta_activate token, wme = nil, assignments = { }
      children.each do |child|
        child.beta_activate Token.new( child, token, token.wme, { @variable => @body.respond_to?(:call) ? @body.call(token) : @body } )
      end
 end
```

This makes the overlay test pass, you probably have a better way though, any thoughts?

**Comment by ulfurinn** [comment]
Please provide a full example demonstrating the issue.

**Comment by jrabramson** [comment]
Just submmited a pull for a test that shows it breaking, here is the gist;

```
production = engine << rule {
      forall {
        has 1, 2, :X
        assign(:Something) { 6 }
      }
      make {
        collect :Something, :stuff
        gen :person, 'stuff', :Something
      }
    }

    engine.with_overlay { |overlay|
      overlay << [1,2,3]
      expect(production).to have(1).token
    }

    expect(production).to have(0).tokens
```

This fails with 1 token on the second expect.

**Comment by ulfurinn** [comment]
Released v0.2.1. It was a wee bit more complicated and will require further work, but this fixes the immediate issue.
